File: wuyou-backmanage-web/src/main/webapp/js/shopservice/goodsService.js
//服务层
app.service('goodsService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../shopGoods/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../shopGoods/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../shopGoods/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../shopGoods/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../shopGoods/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../shopGoods/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../shopGoods/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: wuyou-backmanage-interface/src/main/java/com/wuyou/backmanage/service/GoodsService.java
package com.wuyou.backmanage.service;
import java.util.List;
import com.wuyou.pojo.TbGoods;
import com.wuyou.pojo.TbItem;
import com.wuyou.pojogroup.Goods;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Goods goods);
	
	
	/**
	 * 修改
	 */
	public void update(Goods goods);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Goods findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage2(TbGoods goods, int pageNum,int pageSize);
	
	/**
	 * 批量修改状态
	 * @param ids
	 * @param status
	 */
	public void updateStatus(Long []ids,String status);

	/**
	 * 根据商品ID和状态查询Item表信息  
	 * @param goodsId
	 * @param status
	 * @return
	 */
	public List<TbItem> findItemListByGoodsIdandStatus(Long[] goodsIds, String status );

}
